spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java

```diff
@@ -0,0 +1,192 @@
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableProperties;
+import org.apache.iceberg.hadoop.HadoopInputFile;
+import org.apache.iceberg.io.InputFile;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+import org.apache.iceberg.util.PropertyUtil;
+import org.apache.spark.sql.RuntimeConfig;
+import org.apache.spark.sql.SparkSession;
+
+/**
+ * A class for common Iceberg configs for Spark reads.
+ * <p>
+ * If a config is set at multiple levels, the following order of precedence is used (top to bottom):
+ * <ol>
+ *   <li>Read options</li>
+ *   <li>Session configuration</li>
+ *   <li>Table metadata</li>
+ * </ol>
+ * The most specific value is set in read options and takes precedence over all other configs.
+ * If no read option is provided, this class checks the session configuration for any overrides.
+ * If no applicable value is found in the session configuration, this classes uses the table metadata.
+ * <p>
+ * Note this class is NOT meant to be serialized and sent to executors.
+ */
+public class SparkReadConf {
+
+  private static final Set<String> LOCALITY_WHITELIST_FS = ImmutableSet.of("hdfs");
+
+  private final Table table;
+  private final RuntimeConfig sessionConf;
+  private final Map<String, String> readOptions;
+
+  public SparkReadConf(SparkSession spark, Table table, Map<String, String> readOptions) {
+    this.table = table;
+    this.sessionConf = spark.conf();
+    this.readOptions = readOptions;
+  }
+
+  public boolean localityEnabled() {
+    InputFile file = table.io().newInputFile(table.location());
+
+    if (file instanceof HadoopInputFile) {
+      String scheme = ((HadoopInputFile) file).getFileSystem().getScheme();
+      boolean defaultValue = LOCALITY_WHITELIST_FS.contains(scheme);
+      return PropertyUtil.propertyAsBoolean(
+          readOptions,
+          SparkReadOptions.LOCALITY,
+          defaultValue);
+    }
+
+    return false;
+  }
+
+  public boolean vectorizationEnabled(FileFormat fileFormat) {
+    String readOptionValue = readOptions.get(SparkReadOptions.VECTORIZATION_ENABLED);
+    if (readOptionValue != null) {
+      return Boolean.parseBoolean(readOptionValue);
+    }
+
+    String sessionConfValue = sessionConf.get(SparkSQLConfigs.VECTORIZATION_ENABLED, null);
+    if (sessionConfValue != null) {
+      return Boolean.parseBoolean(sessionConfValue);
+    }
+
+    switch (fileFormat) {
+      case PARQUET:
+        return PropertyUtil.propertyAsBoolean(
+            table.properties(),
+            TableProperties.PARQUET_VECTORIZATION_ENABLED,
+            TableProperties.PARQUET_VECTORIZATION_ENABLED_DEFAULT);
+
+      case ORC:
+        return PropertyUtil.propertyAsBoolean(
+            table.properties(),
+            TableProperties.ORC_VECTORIZATION_ENABLED,
+            TableProperties.ORC_VECTORIZATION_ENABLED_DEFAULT);
+
+      default:
+        return false;
+    }
+  }
+
+  public int batchSize(FileFormat fileFormat) {
```

Comment:
This may be personal preference here, but I'd much rather we have separate batch size getters for Parquet and ORC and put the logic about which to use in the readers themselves. We already have to do this with 

https://github.com/apache/iceberg/pull/3132/files#diff-596268fde5717c19c744ede28de3c9c71d0496fcec746ce774e8a55af25afcfaR363-R369

So why not just have one getter for each property and do the choosing there?

Reply:
This keeps the logic of "Do you support batch reads" in the batch reader rather than in the global configuration class

Reply:
Are you proposing we move `vectorizationEnabled` and `batchSize` out of this class to the readers?
By putting this logic here, I was hoping to reduce the duplication between Spark 2 and 3.

I can introduce separate configs for ORC and Parquet if that seems cleaner.

Reply:
Yes, if you think that isn't a good idea we don't have to. I was just thinking that we should have a 
batchSizeOrc
batchSizeParquet

In the conf, and then the reader would choose them in
```
  private int batchSize(boolean isParquetOnly, boolean isOrcOnly) {
    if (isParquetOnly) {
      return readConf.batchSizeParquet()
    } else if (isOrcOnly) {
      return readConf.batchSizeOrc()
    } else {
      return 0;
```

Instead of the current code which passes through the file type.

Reply:
Sounds good, I'll switch to that.